src/briefcase/integrations/subprocess.py

```diff
@@ -88,12 +88,26 @@ def get_process_id_by_command(
     return None
 
 
+class NativeAppContext:
+    """A wrapper around subprocess for use as an app-bound tool."""
+
+    @classmethod
+    def verify(cls, tools, app):
```

Comment:
This was my desired implementation as well since it is a bit cleaner and decouples the tools from the ToolCache architecture a bit.

However, it means any tool that's passed `tools[app]` will lose access to any already verified "universal tool". This was especially problematic before `Subprocess` and `Download` were added as "default tools". (This was also an impetus for allowing an app-bound `ToolCache` fallback to a universal tool through `__getattr__()`.)

Following that out...in this case, `Docker` would be verified for each app and in the general case, an app-bound tool could be verified up to N+1 times (where N is number of apps).

So, ultimately, I compromised at just passing `tools`.

Reply:
Unless you mean only changing `NativeAppContext` to accept `tools[app]` while `DockerAppContext` would still accept `tools`?

Reply:
The intention would have been a 100% consistent API; however, having access to the "universal" tools seems like enough of an argument to counter any "architectural purity" thoughts.